lib/class-wp-rest-block-types-controller.php

```diff
@@ -0,0 +1,326 @@
+	 */
+	public function prepare_item_for_response( $block_type, $request ) {
+
+		$fields = $this->get_fields_for_response( $request );
+		$data   = array();
+		if ( in_array( 'name', $fields, true ) ) {
+			$data['name'] = $block_type->name;
+		}
+
+		if ( in_array( 'attributes', $fields, true ) ) {
+			$data['attributes'] = $block_type->get_attributes();
+		}
+
+		if ( in_array( 'is_dynamic', $fields, true ) ) {
+			$data['is_dynamic'] = $block_type->is_dynamic();
+		}
+
+		$extra_fields = array( 'editor_script', 'script', 'editor_style', 'style' );
+		foreach ( $extra_fields as $extra_field ) {
+			if ( in_array( $extra_field, $fields, true ) ) {
+				$data[ $extra_field ] = (string) $block_type->$extra_field;
+			}
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		$response = rest_ensure_response( $data );
+
+		$response->add_links(
+			array(
+				'collection'              => array(
+					'href' => rest_url( sprintf( '%s/%s', $this->namespace, $this->rest_base ) ),
+				),
+				'https://api.w.org/items' => array(
+					'href' => rest_url( sprintf( '%s/%s/%s', $this->namespace, $this->rest_base, $block_type->name ) ),
+				),
+			)
+		);
+
+		/**
+		 * Filters a block type returned from the REST API.
+		 *
+		 * Allows modification of the block type data right before it is returned.
+		 *
+		 * @param WP_REST_Response $response   The response object.
+		 * @param object           $block_type The original block type object.
+		 * @param WP_REST_Request  $request    Request used to generate the response.
+		 */
+		return apply_filters( 'rest_prepare_block_type', $response, $block_type, $request );
+	}
+
+	/**
+	 * Retrieves the block type' schema, conforming to JSON Schema.
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'block-type',
+			'type'       => 'object',
+			'properties' => array(
+				'name'          => array(
+					'description' => __( 'Unique name identifying the block type.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'attributes'    => array(
+					'description' => __( 'Block attributes.', 'gutenberg' ),
+					'type'        => 'object',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'is_dynamic'    => array(
+					'description' => __( 'Is the block dynamically rendered.', 'gutenberg' ),
+					'type'        => 'boolean',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'editor_script' => array(
```

Comment:
Here in other places I see snake_case naming pattern. It’s probably the way to go in REST API. It will create friction though since we use camelCase in JavaScript. I guess it’s inevitable.

Reply:
This is the pattern in PHP. Not sure we can do anything.

Reply:
The interesting part is that `attributes` will contain camelCased names, e.g.:
```json
"attributes": {
		"align": {
			"type": "string",
			"enum": [ "left", "center", "right", "wide", "full" ]
		},
		"className": {
			"type": "string"
		},
		"displayAsDropdown": {
			"type": "boolean",
			"default": false
		},
		"showHierarchy": {
			"type": "boolean",
			"default": false
		},
		"showPostCounts": {
			"type": "boolean",
			"default": false
		}
	}
```

Reply:
I checked core and the pattern seems to be snake_case.

https://github.com/WordPress/wordpress-develop/blob/master/src/wp-includes/rest-api/endpoints/class-wp-rest-comments-controller.php#L1396
https://github.com/WordPress/wordpress-develop/blob/master/src/wp-includes/rest-api/endpoints/class-wp-rest-taxonomies-controller.php#L346
https://github.com/WordPress/wordpress-develop/blob/master/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L2046

Reply:
Changed to camelCase in https://github.com/WordPress/gutenberg/pull/21065/commits/3905157a6b9cd342c354f18ee819a6c49c467965

Reply:
As you mentioned, the REST API uses `snake_case`. I'm personally -1 to changing the REST API to start using `camelCase`, but I'm strongly opposed to not following that pattern here as a one-off. If we want to make that change, it should be discussed separately as something to be applied to all future REST API property names.

Reply:
Changed back in https://github.com/WordPress/gutenberg/pull/21065/commits/8f32a723e817486be8bcc390605458feee101b60